modules/network/src/main/java/org/apache/ignite/internal/network/serialization/SerializationService.java

```diff
@@ -100,7 +127,27 @@ public ClassDescriptor getClassDescriptor(int typeDescriptorId) {
      * @param typeName Class' name.
      * @return Class descriptor.
      */
-    public ClassDescriptor getClassDescriptor(String typeName) {
-        return userObjectSerializer.getClassDescriptor(typeName);
+    public ClassDescriptor getOrCreateDescriptor(String typeName) {
+        Class<?> clazz;
+        try {
+            clazz = Class.forName(typeName);
+        } catch (ClassNotFoundException e) {
+            throw new IllegalArgumentException("Class " + typeName + " is not found", e);
```

Comment:
Should a serialization-specific exception be thrown here? It seems possible to encounter such situation.